tc/core/halide2isl.cc

```diff
@@ -453,8 +453,8 @@ isl::schedule makeScheduleTreeHelper(
     std::tie(newReads, newWrites) =
         extractAccesses(iterationDomain, op, accesses);
 
-    *reads = reads->unite(newReads);
-    *writes = writes->unite(newWrites);
+    body->reads = body->reads.unite(newReads);
```

Comment:
maybe we want a function that takes a union on instances of "Body"?

Reply:
> maybe we want a function that takes a union on instances of "Body"?

Then we would have to construct a Body from the relations first.
Since C++ doesn't have designated initializers yet, I think that
would be more error-prone because the relations all have the same type,
so you could easily specify them in the wrong order.

I could change that code to

        body->add_reads(newReads);

if you prefer.  Let me know.

skimo